### PR TITLE
Add codelab docsite

### PR DESCRIPTION
**Pull request by jonathanzong**
For SPAN 2017, we will be hosting a codelab on mdc-web. This PR imports the content of the codelab (live [here](https://codelabs.developers.google.com/codelabs/mdc-web/index.html#0)) as markdown compatible with the material [site generator](https://github.com/material-components/material-components-site-generator). The generated content will have the path `https://material.io/web/codelab/`.

cc @yeelan0319 @willlarche

### CODE REVIEW

**Comment by willlarche** [comment]
Since this codelab is edited for a non-technical audience, it should live in a directory of its own like `docs/codelabs/building-beautiful-sites-simplified`

File: docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+```
+
+You may use the icon font as soon as it loads onto the web page.
+
+## Add the proper mdc-toolbar markup
+
+In `index.html`, replace the `h1` tag with the following markup:
+
+```html
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md]: Have you tried using these codeblocks?

It seems to me these have some weird indentation, not sure if this is something docs introduced.

File: docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  </div>
+</header>
+```
+
+## Add the custom mdc-toolbar styles
+
+Add the following styles to `app.css`:
+
+```css
+#shrine-header {
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md]: Same thing here.

File: docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Alter the `<a id="shrine-nav-icon" ...>` element to look like this:
+
+```html
+<a id="shrine-nav-icon" class="material-icons mdc-ripple-surface"  
+   href="#"
+   aria-label="Click to show the navigation menu"
+   aria-controls="shrine-nav-menu"
+   data-mdc-auto-init="MDCRipple"
+   data-mdc-ripple-is-unbounded>menu</a>
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/3-toolbar.md]: Same thing here.

File: docs/codelabs/building-beautiful-sites/docsite/4-navigation.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Now that our app is set up to use JavaScript, adding a navigation drawer should be simple. Here's the mock our designer has given us for the navigation drawer:
+
+Let's add this to our app. Use the Temporary Drawer from the [mdc-drawer](https://github.com/material-components/material-components-web/tree/master/packages/mdc-drawer) package in order to achieve this functionality.
+
+## Add the drawer markup
+
+Add the markup below for the temporary drawer in `index.html`, right below the markup for the toolbar and before the `<main>` element:
+
+```html
+<aside id="shrine-nav-menu" class="mdc-temporary-drawer" data-mdc-auto-init="MDCTemporaryDrawer">
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/4-navigation.md]: Same thing here.

File: docs/codelabs/building-beautiful-sites/docsite/4-navigation.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      <a class="mdc-list-item" href="#">Clothing</a>
+      <a class="mdc-list-item" href="#">Popsicles</a>
+    </nav>
+  </nav>
+</aside>
+```
+
+## Add the drawer custom styles
+
+Add the following to `app.css` to style the drawer to reflect the mocks:
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/4-navigation.md]: Same thing here. And following codeblocks

File: docs/codelabs/building-beautiful-sites/docsite/4-navigation.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+<script>
+  mdc.autoInit();
+  document.getElementById('shrine-nav-icon').addEventListener('click', function(evt) {
+    evt.preventDefault();
+    document.getElementById('shrine-nav-menu').MDCTemporaryDrawer.open = true;
+  });
+</script>
+```
+
+> Note that the MDCTemporaryDrawer property on the shrine-nav-menu element is the automatically instantiated MDCTemporaryDrawer instance, attached as part of mdc-auto-init.
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/4-navigation.md]: MDCTemporaryDrawer -> `MDCTemporaryDrawer`
shrine-nav-menu -> `shrine-nav-menu`
mdc-auto-init -> `mdc-auto-init`

File: docs/codelabs/building-beautiful-sites/docsite/2-explore.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    <div class="mdc-card shrine-product-card">
+      <section class="mdc-card__primary">
+        <span class="mdc-card__title shrine-product-card__price">$20</span>
+      </section>
+      <img class="shrine-product-card__image" width="240" height="240" alt="Sunglasses" src="assets/sunnies.png">
+    </div>
+    <!-- ... -->
+```
+
+The first thing to notice here is that we’re using a `<link>` tag to load a fully-built version of MDC-Web via the [unpkg CDN](https://unpkg.com/#/). We publish MDC-Web via NPM. Using unpkg is a great way to experiment with MDC-Web without the need to immediately include it as a dependency. Note that we include [normalize.css](https://necolas.github.io/normalize.css/) for consistent cross-browser rendering, as well as the [Roboto font](https://fonts.google.com/specimen/Roboto) from Google Fonts.
```
Comment (yeelan0319) [inline docs/codelabs/building-beautiful-sites/docsite/2-explore.md]: Consider writing: 
Note that we pinned to use version `0.9.1` to the purpose of this codelab as the `latest` version is under active development. For the available components and latest APIs, please checkout our [Github Repository](https://github.com/material-components/material-components-web) and [Demo site](http://material-components-web.appspot.com/).

File: docs/codelabs/building-beautiful-sites/docsite/introduction.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+- Card
+- Toolbar
+- Ripple
+- Drawer
+- List
+- Layout Grid
+
+## What you’ll need
+
+- A recent version of [NodeJS](https://nodejs.org/en/) (which comes bundled with [npm](https://www.npmjs.com/), a JavaScript package manager).
```
[the remark this answers is not in the document]
Comment (willlarche) [reply docs/codelabs/building-beautiful-sites/docsite/introduction.md]: We definitely will. I'm going over the content tonight to simplify.

**Comment by jonathanzong** [comment]
> Since this codelab is edited for a non-technical audience, it should live in a directory of its own like docs/codelabs/building-beautiful-sites-simplified

@willlarche : The issue I'm running into with the route is the generated nav list on the left will include the entire path structure. (so for `/web/docs/codelabs/building-beautiful-sites-simplified` it would look like e.g. `docs > codelabs > Building Beautiful Sites > Add a toolbar`) I think for now—at least for SPAN, where the participants don't need to be aware of other versions—we could keep it as `/web/codelab/` and then when we start adding other platforms/versions, add new routes. What are your thoughts?

**Comment by willlarche** [comment]
@jonathanzong Then let's put comments in these to say "This is a simplified version of Building Beautiful Sites with MDC web edited for a non-technical audience."

**Pull request merged**